### ISSUE TITLE
[App]: Duplicate runs in a sweep

### ISSUE THREAD

**Issue by kbkartik** [wandb/client#3467]
### Current Behavior

As seen in the image below, we see 3 runs with the same name (`RN50_strategy_feature_extraction_data_aug_True`) when doing a bayesian sweep over config parameters:

![image](https://user-images.githubusercontent.com/79010023/161521983-258e9f41-37c1-43ac-a4cc-c8888de896c5.png)

### Expected Behavior

Duplicate runs shouldn't be computed under bayesian sweep.

### Steps To Reproduce

1. Use the following configs: 
`bayes_config = {
    "name" : "PartB_instance1",
    "method" : "bayes",
    "metric" : {
        "name" : "val_acc",
        "goal" : "maximize",
    },
    "parameters" : {
        "model_name" : {
            "values" : [['RN50', (224, 224)], 
                        ['IV3', (299, 299)], 
                        ['MV3S', (224, 224)]]
        },
        "strategy" : {
            "values" : ['finetuning', 'feature_extraction']
        },
        "data_aug" : {
            "values" : [True, False]
        },
    }
}`

2. Test the sweep on some random NN:

`sweep_id = wandb.sweep(bayes_config, project="test", entity="abc")`
`wandb.agent(sweep_id, function=single_sweep, count=12)`

3. There are 12 combinations for the above config parameters which means one should have 12 unique runs.

### Environment

OS: Linux

Browsers: Chrome

Version: I guess the latest one. I am doing pip install on Colab

**Comment by nate-wandb** [comment]
Hi @kbkartik, when there is a small hyper-parameter search space it can happen that a Bayesian search can recommend a duplicate of the same parameters if some of the hyper-parameters have proven unlikely to improve the goal metric. 

We're currently working to handle this special case better and I can notify you once a fix has been implemented. In the meantime I would recommend using a grid search if you would like to explore all 12 hyper-parameter combinations. 

Let me know if this answers your question or if I can help clarify anything.
Thank you,
Nate

**Comment by nick-maykr** [comment]
@nate-wandb any progress?
Just wasted some resources on a long running compute to find this out. How is this not an important issue? A run with the same parameters should never be run twice, regardless the search algorithm.

**Comment by nate-wandb** [comment]
Hi @nick-maykr, sorry to hear this. I can raise the priority on this. Could you share your parameter space so we can take a look? Also, do you happen to be running multiple agents in parallel?

**Comment by sanesma93** [comment]
Hello, I am experiencing the same problem my hyperparameters are these
`'KLDgamma': {'values': [0.3, 0.5, 0.75, 1.0]},
 'experiment': {'values': ['TrueClass','TopClass','AllClasses']}`
I've had to manually stop individual runs until it starts a run that doesn't use hyperparameter values that have already been tried
![image](https://github.com/wandb/wandb/assets/15927795/f1744940-4004-4bbe-b262-7a47782175c6)

**Comment by sanesma93** [comment]
Hello,
This is from a single agent

Samantha Escobar Martínez BSc
(she/they)

**Comment by nate-wandb** [comment]
Hi @sanesma93, could you let me know what version of `wandb` you are using? This should be resolved in the latest version `0.16.0` if you have a chance to try this out

**Comment by volgasezen** [comment]
> Hi @sanesma93, could you let me know what version of `wandb` you are using? This should be resolved in the latest version `0.16.0` if you have a chance to try this out

While I'm not the original issuer, having run about 30 sweeps across a parameter grid of size 256 I've seen reduced number of duplicates upon upgrading to 0.16 (was on 0.15.8 before), but I still caught a few. Also running a bayesian search.

Specifically my grid was the following: 

`
sweep_config = {
    "method": "bayes",
    "metric": {
      "name": "val_loss",
      "goal": "minimize"   
    },
    "parameters": {
        "num_layers": {
            "values": [2,3,4,5]
        },
        "hidden_size": {
            "values": [4,8,16,32]
        },
        "hidden_ln": {
            "values": [32,64,128,256]
        },
        "drop": {
            "values": [0,0.1,0.2,0.4]
        }
    },
    "early_terminate": {
        "type": "hyperband",
        "eta": 2,
        "min_iter":2
     }
}`

![image](https://github.com/wandb/wandb/assets/127928023/bd67f79d-19c2-472b-b8a7-97b2c4a2b199)

The phenomenon mentioned by @nate-wandb may still be the case, though I am curious about the effect of the marginal update when the run is duplicated. It may just be an attempt to perturb the overall run given new information from previous non-duplicate runs, but please correct if I am wrong.

**Comment by nate-wandb** [comment]
Hi @volgasezen, thank you for testing `0.16.0` and sorry to hear this is still the case. I'll pass this on to the engineering team and let them know this is still the case. I'll test out your config as well and see if I can repro this on my side.

**Comment by mikeogezi** [comment]
I'm still having this issue with version 0.16.3, running a single agent.

**Comment by caixd-220529** [comment]
Same issue
